bson/src/main/org/bson/codecs/configuration/mapper/ClassModelCodecProvider.java

```diff
@@ -52,12 +60,17 @@ public static ProviderBuilder builder() {
     @Override
     @SuppressWarnings({"unchecked", "rawtypes"})
     public <T> Codec<T> get(final Class<T> clazz, final CodecRegistry registry) {
-        Codec<T> codec = null;
+        ClassModelCodec<?> codec = null;
         if (registered.contains(clazz)) {
-            final ClassModel model = new ClassModel(registry, resolver, (Class<Object>) clazz);
-            codec = (Codec<T>) new ClassModelCodec(model);
+            codec = mapped.get(clazz);
+            if (codec == null && conventionPack.isMappable(clazz)) {
```

Comment:
It just seems completely redundant to me. Even in the MorphiaCodecProvider, presumably it wouldn't recognize things that morphia couldn't map at the codec provider level... hence, the isMappable check is still unnecessary.